crates/extension/src/wasm_host.rs

```diff
@@ -159,29 +159,25 @@ impl WasmHost {
     }
 
     async fn build_wasi_ctx(&self, manifest: &Arc<ExtensionManifest>) -> Result<wasi::WasiCtx> {
-        use cap_std::{ambient_authority, fs::Dir};
-
         let extension_work_dir = self.work_dir.join(manifest.id.as_ref());
         self.fs
             .create_dir(&extension_work_dir)
             .await
             .context("failed to create extension work dir")?;
 
-        let work_dir_preopen = Dir::open_ambient_dir(&extension_work_dir, ambient_authority())
-            .context("failed to preopen extension work directory")?;
-        let current_dir_preopen = work_dir_preopen
-            .try_clone()
-            .context("failed to preopen extension current directory")?;
-        let extension_work_dir = extension_work_dir.to_string_lossy();
-
-        let perms = wasi::FilePerms::all();
+        let file_perms = wasi::FilePerms::all();
         let dir_perms = wasi::DirPerms::all();
 
         Ok(wasi::WasiCtxBuilder::new()
             .inherit_stdio()
-            .preopened_dir(current_dir_preopen, dir_perms, perms, ".")
-            .preopened_dir(work_dir_preopen, dir_perms, perms, &extension_work_dir)
-            .env("PWD", &extension_work_dir)
+            .preopened_dir(&extension_work_dir, ".", dir_perms, file_perms)?
```

Comment:
`preopened_dir` now takes an `AsRef<Path>` for the host directory, so we don't need to call `Dir::open_ambient_dir` ourselves.

The order of the parameters also changed a bit, with the `guest_path` moving from the last parameter to the second.

For reference:

https://github.com/bytecodealliance/wasmtime/pull/8228/files#diff-e590d3fac47f1511c34190cc30c0d5488c555844c118be047feba373e108d420L100-R100